Customize the file browser toolbar via the settings (#12281)
jupyterlab/jupyterlab: docs/source/extension/extension_points.rst

.. literalinclude:: ../snippets/packages/settingregistry/src/toolbarItem.json
   :language: json

Generic Widget with Toolbar
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The logic explain in the previous paragraph can be used to customize any widgets with a toolbar.

Reviewer: ```suggestion
The logic detailed in the previous section can be used to customize any widgets with a toolbar.
```

The additional keys used in ``jupyter.lab.toolbars`` settings attributes are:

- ``FileBrowser``: Default file browser panel toolbar items

> To create the toolbar items from that definition on a widget would can use a logic like the following:

Reviewer: ```suggestion
Here is an example for enabling that definition on a widget:
```

.. code:: typescript

   function activatePlugin(
     app: JupyterFrontEnd,
     // ...
     toolbarRegistry: IToolbarWidgetRegistry,
     settingRegistry: ISettingRegistry
   ): void {

     const browser = new FileBrowser();

     // Toolbar

>      // - Define a customized toolbar item

Reviewer: ```suggestion
     // - Define a custom toolbar item
```
